File: SDKs/googleapis.com/servicebroker/v1alpha1/go/pkg/models/operations/servicebrokerprojectsbrokersv2serviceinstancesgetlastoperation.go
package operations

import (
"openapi/pkg/models/shared")

type ServicebrokerProjectsBrokersV2ServiceInstancesGetLastOperationPathParams struct {
    InstanceID string `pathParam:"style=simple,explode=false,name=instanceId"`
    Parent string `pathParam:"style=simple,explode=false,name=parent"`
    
}

type ServicebrokerProjectsBrokersV2ServiceInstancesGetLastOperationQueryParams struct {
    DollarXgafv *shared.XgafvEnum `queryParam:"style=form,explode=true,name=$.xgafv"`
    AccessToken *string `queryParam:"style=form,explode=true,name=access_token"`
    Alt *shared.AltEnum `queryParam:"style=form,explode=true,name=alt"`
    Callback *string `queryParam:"style=form,explode=true,name=callback"`
    Fields *string `queryParam:"style=form,explode=true,name=fields"`
    Key *string `queryParam:"style=form,explode=true,name=key"`
    OauthToken *string `queryParam:"style=form,explode=true,name=oauth_token"`
    Operation *string `queryParam:"style=form,explode=true,name=operation"`
    PlanID *string `queryParam:"style=form,explode=true,name=planId"`
    PrettyPrint *bool `queryParam:"style=form,explode=true,name=prettyPrint"`
    QuotaUser *string `queryParam:"style=form,explode=true,name=quotaUser"`
    ServiceID *string `queryParam:"style=form,explode=true,name=serviceId"`
    UploadType *string `queryParam:"style=form,explode=true,name=uploadType"`
    UploadProtocol *string `queryParam:"style=form,explode=true,name=upload_protocol"`
    
}

type ServicebrokerProjectsBrokersV2ServiceInstancesGetLastOperationSecurity struct {
    Oauth2 shared.SchemeOauth2 `security:"scheme,type=oauth2"`
    Oauth2c shared.SchemeOauth2c `security:"scheme,type=oauth2"`
    
}

type ServicebrokerProjectsBrokersV2ServiceInstancesGetLastOperationRequest struct {
    PathParams ServicebrokerProjectsBrokersV2ServiceInstancesGetLastOperationPathParams 
    QueryParams ServicebrokerProjectsBrokersV2ServiceInstancesGetLastOperationQueryParams 
    Security ServicebrokerProjectsBrokersV2ServiceInstancesGetLastOperationSecurity 
    
}

type ServicebrokerProjectsBrokersV2ServiceInstancesGetLastOperationResponse struct {
    Body []byte 
    ContentType string 
    StatusCode int64 
    
}

File: SDKs/googleapis.com/servicebroker/v1alpha1/typescript/src/sdk/models/operations/servicebrokerprojectsbrokersv2serviceinstancesgetlastoperation.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class ServicebrokerProjectsBrokersV2ServiceInstancesGetLastOperationPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=instanceId" })
  instanceId: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=parent" })
  parent: string;
}


export class ServicebrokerProjectsBrokersV2ServiceInstancesGetLastOperationQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=$.xgafv" })
  dollarXgafv?: shared.XgafvEnum;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=access_token" })
  accessToken?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=alt" })
  alt?: shared.AltEnum;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=callback" })
  callback?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=fields" })
  fields?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=key" })
  key?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=oauth_token" })
  oauthToken?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=operation" })
  operation?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=planId" })
  planId?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=prettyPrint" })
  prettyPrint?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=quotaUser" })
  quotaUser?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=serviceId" })
  serviceId?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=uploadType" })
  uploadType?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=upload_protocol" })
  uploadProtocol?: string;
}


export class ServicebrokerProjectsBrokersV2ServiceInstancesGetLastOperationSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  oauth2: shared.SchemeOauth2;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  oauth2c: shared.SchemeOauth2c;
}


export class ServicebrokerProjectsBrokersV2ServiceInstancesGetLastOperationRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: ServicebrokerProjectsBrokersV2ServiceInstancesGetLastOperationPathParams;

  @SpeakeasyMetadata()
  queryParams: ServicebrokerProjectsBrokersV2ServiceInstancesGetLastOperationQueryParams;

  @SpeakeasyMetadata()
  security: ServicebrokerProjectsBrokersV2ServiceInstancesGetLastOperationSecurity;
}


export class ServicebrokerProjectsBrokersV2ServiceInstancesGetLastOperationResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  body?: Uint8Array;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
